executor/oci/resolvconf.go

```diff
@@ -83,19 +99,24 @@ func GetResolvConf(ctx context.Context, stateDir string, idmap *idtools.Identity
 				dnsOptions = resolvconf.GetOptions(dt)
 			}
 
-			f, err = resolvconf.Build(tmpPath, dnsNameservers, dnsSearchDomains, dnsOptions)
+			f, err := resolvconf.Build(tmpPath, dnsNameservers, dnsSearchDomains, dnsOptions)
 			if err != nil {
 				return struct{}{}, err
 			}
 			dt = f.Content
 		}
 
-		f, err = resolvconf.FilterResolvDNS(dt, true)
```

Comment:
Instead of making changes upstream to keep local ns we can just bring minimal changes in BuildKit. As discussed with @thaJeztah and @robmry, changes are currently made to improve `libnetwork/resolveconf` pkg in moby related tohttps://github.com/moby/moby/issues/46968

Reply:
The alternative would be to create a copy of the host's `/etc/resolv.conf` (or whatever it's symlinked to) and use that as-is.

Effectively, when running with `host` networking, the container's network _IS_ the host's network, so it's within line of expectations that it's configured identically.

Reply:
> Instead of making changes upstream to keep loca

What are the changes that would be needed? I think we can just call "filter" (that really is filter localhost and add default) only if `!host || len(Nameservers()) == 0`

Reply:
Per discussion on slack with @robmry it looks indeed that for `host` networking, we can just take a copy of the host's `/etc/resolv.conf` without any modifications. "bridge" network likely needs the existing code (filter out loopback addresses); that is, unless we consider using an embedded DNS at some point, but that's not the case (yet)

Reply:
> we can just take a copy of the host's /etc/resolv.conf without any modifications.

Unless it doesn't have any nameservers defined.

Reply:
If there are no nameservers in the host's resolv.conf, for host networking, libnetwork doesn't add the defaults. The nameservers would have to come from a CLI option in that case. For other networks, it would add the defaults.

No idea if that was a deliberate decision or just something that happened, but maybe build should do the same?

(So, that would mean just copying the file for host networking, or passing it through FilterResolvDNS() for other networks.)

Reply:
> Unless it doesn't have any nameservers defined.

Wouldn't that be a broken host? I guess in that situation many more things would be broken

Reply:
> Wouldn't that be a broken host?

Not necessarily. Buildkit can run in weird setups, not actual full Linux host and apps can have their own DNS config.

Reply:
Yes, I think it'd be unusual! But not adding defaults in that case would be consistent with normal containers, and make the code here a bit simpler.

Reply:
> Not necessarily. Buildkit can run in weird setups, not actual full Linux host and apps can have their own DNS config.

Sorry, hadn't seen that when I replied.

And, now I see FilterResolvDNS() was always called before, so it'd be a change in behaviour to not-add the defaults in that case. We could add a function to 'resolvconf' to add the defaults if-missing, that'd save duplicating the defaults here - but maybe the dependency between projects makes that tricky (I'm not sure how the releases fit together).

Reply:
Doh, I confused myself ... FilterResolvDNS() will do the trick if there are no nameservers, so there's no need to have the defaults here. It's exactly as you said ...

> I think we can just call "filter" (that really is filter localhost and add default) only if !host || len(Nameservers()) == 0